test/libsolidity/SolidityExecutionFramework.cpp

```diff
@@ -61,18 +61,40 @@ bytes SolidityExecutionFramework::multiSourceCompileContract(
 	evmasm::LinkerObject obj;
 	if (m_compileViaYul)
 	{
-		yul::AssemblyStack asmStack(
-			m_evmVersion,
-			yul::AssemblyStack::Language::StrictAssembly,
-			// Ignore optimiser settings here because we need Yul optimisation to
-			// get code that does not exhaust the stack.
-			OptimiserSettings::full()
-		);
-		bool analysisSuccessful = asmStack.parseAndAnalyze("", m_compiler.yulIROptimized(contractName));
-		solAssert(analysisSuccessful, "Code that passed analysis in CompilerStack can't have errors");
+		// Try compiling twice: If the first run fails due to stack errors, forcefully enable
+		// the optimizer.
+		for (int i = 0; i < 2; ++i)
```

Comment:
```suggestion
		for (bool forceEnableOptimizer: {false, true})
```
Might be a bit more descriptive...